### PR TITLE
deseriaize variable on dicts or vectors

### PR DESCRIPTION
**Pull request by dehann**
Should standardize to dictionary versions only, i.e. `ppeDict` and `solverDataDict`.  Should not have vector versions under ppes or solverData.

### CODE REVIEW

File: test/test_defVariable.jl [hunk 1]
```diff
@@ -40,10 +40,10 @@ struct NotAManifold end
 @test getDimension(TestVarType2()) === 6
 
 @test getPointType(TestVarType1()) == Vector{Float64}
-@test getPointType(TestVarType2()) == ProductRepr{Tuple{Vector{Float64}, Matrix{Float64}}}
+# @test getPointType(TestVarType2()) == ProductRepr{Tuple{Vector{Float64}, Matrix{Float64}}}
 
 @test getPointIdentity(TestVarType1()) == zeros(3)
-@test getPointIdentity(TestVarType2()).parts[1] == ProductRepr(zeros(3), diagm(ones(3))).parts[1]
-@test getPointIdentity(TestVarType2()).parts[2] == ProductRepr(zeros(3), diagm(ones(3))).parts[2]
+@test getPointIdentity(TestVarType2())[1] == ArrayPartition(zeros(3), diagm(ones(3)))[1]
```
Comment (Affie) [inline test/test_defVariable.jl:46]: This should be .x in place of .parts. Or use submanifold_components

Comment (dehann) [reply test/test_defVariable.jl:46]: ah thanks

File: test/test_defVariable.jl [hunk 2]
```diff
@@ -24,11 +24,11 @@ struct NotAManifold end
 @test getDimension(TestVarType2) === 6
 
 @test getPointType(TestVarType1) == Vector{Float64}
-@test getPointType(TestVarType2) == ProductRepr{Tuple{Vector{Float64}, Matrix{Float64}}}
+# @test getPointType(TestVarType2) == ProductRepr{Tuple{Vector{Float64}, Matrix{Float64}}}
 
 @test getPointIdentity(TestVarType1) == zeros(3)
-@test getPointIdentity(TestVarType2).parts[1] == ProductRepr(zeros(3), diagm(ones(3))).parts[1]
-@test getPointIdentity(TestVarType2).parts[2] == ProductRepr(zeros(3), diagm(ones(3))).parts[2]
+@test getPointIdentity(TestVarType2)[1] == ArrayPartition(zeros(3), diagm(ones(3)))[1]
```
Comment (dehann) [inline test/test_defVariable.jl:30]:
```suggestion
@test getPointIdentity(TestVarType2).x[1] == ArrayPartition(zeros(3), diagm(ones(3))).x[1]
```

File: test/test_defVariable.jl [hunk 3]
```diff
@@ -24,11 +24,11 @@ struct NotAManifold end
...
+@test getPointIdentity(TestVarType2)[2] == ArrayPartition(zeros(3), diagm(ones(3)))[2]
```
Comment (dehann) [inline test/test_defVariable.jl:31]:
```suggestion
@test getPointIdentity(TestVarType2).x[2] == ArrayPartition(zeros(3), diagm(ones(3))).x[2]
```

File: test/test_defVariable.jl (hunk 1 above)
Comment (dehann) [reply test/test_defVariable.jl:46]:
```suggestion
@test getPointIdentity(TestVarType2()).x[1] == ArrayPartition(zeros(3), diagm(ones(3))).x[1]
```

File: test/test_defVariable.jl [hunk 4]
```diff
@@ -40,10 +40,10 @@ struct NotAManifold end
...
 
 @test getPointType(TestVarType1()) == Vector{Float64}
-@test getPointType(TestVarType2()) == ProductRepr{Tuple{Vector{Float64}, Matrix{Float64}}}
+# @test getPointType(TestVarType2()) == ProductRepr{Tuple{Vector{Float64}, Matrix{Float64}}}
 
 @test getPointIdentity(TestVarType1()) == zeros(3)
-@test getPointIdentity(TestVarType2()).parts[1] == ProductRepr(zeros(3), diagm(ones(3))).parts[1]
-@test getPointIdentity(TestVarType2()).parts[2] == ProductRepr(zeros(3), diagm(ones(3))).parts[2]
+@test getPointIdentity(TestVarType2())[1] == ArrayPartition(zeros(3), diagm(ones(3)))[1]
+@test getPointIdentity(TestVarType2())[2] == ArrayPartition(zeros(3), diagm(ones(3)))[2]
```
Comment (dehann) [inline test/test_defVariable.jl:47]:
```suggestion
@test getPointIdentity(TestVarType2()).x[2] == ArrayPartition(zeros(3), diagm(ones(3))).x[2]
```

File: test/test_defVariable.jl [hunk 5]
```diff
@@ -40,10 +40,10 @@ struct NotAManifold end
 @test getDimension(TestVarType2()) === 6
 
 @test getPointType(TestVarType1()) == Vector{Float64}
-@test getPointType(TestVarType2()) == ProductRepr{Tuple{Vector{Float64}, Matrix{Float64}}}
+# @test getPointType(TestVarType2()) == ProductRepr{Tuple{Vector{Float64}, Matrix{Float64}}}
```
Comment (Affie) [inline test/test_defVariable.jl:43]:
```suggestion
# @test getPointType(TestVarType2()) == ArrayPartition{Float64, Tuple{Vector{Float64}, Matrix{Float64}}}
```

File: test/test_defVariable.jl [hunk 6]
```diff
@@ -24,11 +24,11 @@ struct NotAManifold end
 @test getDimension(TestVarType2) === 6
 
 @test getPointType(TestVarType1) == Vector{Float64}
-@test getPointType(TestVarType2) == ProductRepr{Tuple{Vector{Float64}, Matrix{Float64}}}
+# @test getPointType(TestVarType2) == ProductRepr{Tuple{Vector{Float64}, Matrix{Float64}}}
```
Comment (Affie) [inline test/test_defVariable.jl:27]:
```suggestion
@test getPointType(TestVarType2) == ArrayPartition{FLoat64,Tuple{Vector{Float64}, Matrix{Float64}}}
```

**Pull request merged**